### ISSUE TITLE
`surfaceTint`, field of `ColorScheme`, can’t be overwritten

### ISSUE THREAD

**Issue by epentesi** [flutter/flutter#104350]
## Steps to Reproduce

1. Make a custom `ColorScheme` named `customColorScheme` using `ColorScheme.fromSeed()` 
2. The `seedColor`is `Colors.orange`
3. Set `surfaceTint` to `Colors.green` using `copyWith()`
```dart
ColorScheme customColorScheme = ColorScheme.fromSeed(seedColor: Colors.orange)
.copyWith(
  surfaceTint: Colors.green,
);
```
4. Make a `Container` with `color: customColorScheme.surfaceTint`
```dart
Container(
  width: 100,
  height: 100,
  color: customColorScheme.surfaceTint,
)
```
**Expected results:** 
A Container filled with `Colors.green` color

**Actual results:** 
A Container filled with `customColorScheme.primary` color (tone 40 of `Colors.orange` used as seed)

## Steps to solve the bug:
1. Open `color_scheme.dart` from `flutter/packages/flutter/lib/src/material/`
2. Change line 840 from
```surfaceTint: _surfaceTint ?? this.surfaceTint,```
to
```surfaceTint: surfaceTint ?? this.surfaceTint,```

**Comment by huycozy** [comment]
Hi, I'm able to reproduce this issue on the latest `stable` and `master` channels.
This issue is only happening on `ColorScheme.fromSeed()` constructor and does not happen other constructors, such as: `ColorScheme.light`, `ColorScheme.dark`.

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  ColorScheme customColorScheme = ColorScheme.fromSeed(seedColor: Colors.orange)
      .copyWith(
    surfaceTint: Colors.green,
  );

  ColorScheme customColorScheme2 = ColorScheme.light(surfaceTint: Colors.green);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Flutter Demo'),
      ),
      floatingActionButton: FloatingActionButton(
        child: const Icon(Icons.save),
        onPressed: () async {
        },
      ),
      body:  Center(
        child:  Container(
          width: 100,
          height: 100,
          color: customColorScheme.surfaceTint,
        )
      ),
    );
  }

}
```
</details>

<details>
<summary>Demo</summary>

![104350](https://user-images.githubusercontent.com/104349824/169802512-794d9636-b018-4b46-a483-28ac42ed1676.PNG)

</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[√] Flutter (Channel stable, 3.0.1, on Microsoft Windows [Version 10.0.19043.1706], locale en-US)
    • Flutter version 3.0.1 at C:\WIP\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (4 days ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

Checking Android licenses is taking an unexpectedly long time...[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = D:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32421.90
    • Windows 10 SDK version 10.0.20348.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.67.1)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.40.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1706]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 101.0.4951.67
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 98.0.1108.56

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[√] Flutter (Channel master, 3.1.0-0.0.pre.854, on Microsoft Windows [Version 10.0.19043.1706], locale en-US)
    • Flutter version 3.1.0-0.0.pre.854 at C:\WIP\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ec20ea80ad (2 days ago), 2022-05-21 16:28:06 -0400
    • Engine revision 1bfe551d6b
    • Dart version 2.18.0 (build 2.18.0-142.0.dev)
    • DevTools version 2.13.1

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = D:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32421.90
    • Windows 10 SDK version 10.0.20348.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.67.1)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.40.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1706]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 101.0.4951.67
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 98.0.1108.56

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by epentesi** [comment]
Hi @huycozy 
The issue happens with the method `copyWith()`.
And it happens even with `light()` and `dark()` constructors.

## Steps to Reproduce

1. I created 3 custom `ColorScheme`s:

- customColorSchemeFromSeedOrange
- customColorSchemeLight
- customColorSchemeDark

2. I applied `surfaceTint: Colors.green` through the `copyWith()` method.
3. I made 3 Containers with their color set to:

- `customColorSchemeFromSeedOrange.surfaceTint`
- `customColorSchemeLight.surfaceTint`
- `customColorSchemeDark.surfaceTint`

### Expected results:
All 3 Containers should be green.

### Actual results:
Every Container is filled according to a tone of the primary color of its own ColorScheme.

<details>
<summary>Code sample</summary>


```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  ColorScheme customColorSchemeFromSeedOrange =
      ColorScheme.fromSeed(seedColor: Colors.orange).copyWith(
    surfaceTint: Colors.green,
  );

  ColorScheme customColorSchemeLight = const ColorScheme.light().copyWith(
    surfaceTint: Colors.green,
  );

  ColorScheme customColorSchemeDark = const ColorScheme.dark().copyWith(
    surfaceTint: Colors.green,
  );

  TextStyle textStyle = const TextStyle(
    color: Colors.white,
  );

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Flutter Demo'),
      ),
      floatingActionButton: FloatingActionButton(
        child: const Icon(Icons.save),
        onPressed: () async {},
      ),
      body: Center(
          child: Column(
        children: [
          Container(
            width: 300,
            height: 100,
            padding: const EdgeInsets.all(16.0),
            color: customColorSchemeFromSeedOrange.surfaceTint,
            child: Text(
              'customColorSchemeFromSeedOrange\nsurfaceTint: Colors.green',
              style: textStyle,
            ),
          ),
          Container(
            width: 300,
            height: 100,
            padding: const EdgeInsets.all(16.0),
            color: customColorSchemeLight.surfaceTint,
            child: Text(
              'customColorSchemeLight\nsurfaceTint: Colors.green',
              style: textStyle,
            ),
          ),
          Container(
            width: 300,
            height: 100,
            padding: const EdgeInsets.all(16.0),
            color: customColorSchemeDark.surfaceTint,
            child: Text(
              'customColorSchemeDark\nsurfaceTint: Colors.green',
              style: textStyle,
            ),
          ),
        ],
      )),
    );
  }
}

```

</details>

Here’s a demo:

<details>
  <summary>Demo</summary>

![demo](https://user-images.githubusercontent.com/27135013/169818181-9a036b89-699d-475b-948f-0fcc4c535d59.png)


</details>

**Comment by darrenaustin** [comment]
@epentesi thanks for the thorough report. Sorry we missed this typo in our reviews. This should be resolved when #104435 lands.
